gossip/src/contact_info_ffi.rs

```diff
@@ -0,0 +1,945 @@
+use {
+    crate::contact_info::ContactInfo,
+    solana_client::connection_cache::Protocol,
+    solana_sdk::pubkey::Pubkey,
+    std::net::{IpAddr, Ipv4Addr, Ipv6Addr, SocketAddr},
+    thiserror::Error,
+};
+
+#[repr(C)]
```

Comment:
probably best to give this an explicit `u` type (ie. `u32`)?
even with `repr(C)` here, if we added enough variants then it could become something new.